narwhal/primary/src/synchronizer.rs

```diff
@@ -339,12 +371,28 @@ impl Synchronizer {
             }
         });
 
-        // Start a task to update gc_round and gc in-memory data.
+        // Start a task to update gc_round, gc in-memory data, and trigger certificate catchup
+        // if no gc / commit happened for 10s.
         let weak_inner = Arc::downgrade(&inner);
         spawn_monitored_task!(async move {
+            const FETCH_TRIGGER_TIMEOUT: Duration = Duration::from_secs(30);
```

Comment:
Do you think this parameter would visibly impact performance?
If so, should it eventually be a config parameters?

Reply:
If I understand correctly we hope this will never / rarely happen? If so we might want to emit a `warn` log line to detect it, or update a metric?

Reply:
> Do you think this parameter would visibly impact performance?
> If so, should it eventually be a config parameters?

Definitely good practice having it as config parameter, although 30 seconds seem quite graceful to me to not affect performance

> If I understand correctly we hope this will never / rarely happen? If so we might want to emit a warn log line to detect it, or update a metric?

Agreed on logging and metric (or use some sort of tag on existing metric to differentiate the command/source that triggered it)

Reply:
Yes the goal of triggering fetching when there is no consensus commit (as opposed to a constant interval), is to rarely activate the codepath and not affect performance in the normal case. Added metric and logging here.